### DISCUSSION TITLE
The constructor is undefined

### DISCUSSION THREAD

**Discussion by deanwean** [FabricMC/fabric-api#2012] [Mod Dev Support] [answered]
### The constructor ZombieAttackGoal(WrogglerEntity, double, boolean) is undefinedJava(134217858)
This is the error I'm having, I've tried a few things and it still didn't work.
I'm trying to give my entity an attack goal so it attacks the player, however each time I try to I get an error.

This is the part where I try to give it the "ZombieAttackGoal" so it would attack the player, it didn't work (it gave me the error in the first sentence), so I changed it to "WrogglerAttackGoal" and added some code below to try to make it work.
```java
protected void initCustomGoals() {
         this.goalSelector.add(2, new WrogglerAttackGoal(this, 1.0D, false)); //This is the attack goal
        this.goalSelector.add(7, new WanderAroundFarGoal(this, 1.0D));
        this.targetSelector.add(2, new FollowTargetGoal<>(this, PlayerEntity.class, true));
        this.targetSelector.add(3, new FollowTargetGoal<>(this, MerchantEntity.class, false));
        this.targetSelector.add(3, new FollowTargetGoal<>(this, IronGolemEntity.class, true));
    }
```

This is the code I added, however it still creates the exact same error.
```java
static class WrogglerAttackGoal extends ZombieAttackGoal {
        public WrogglerAttackGoal(WrogglerEntity wroggler, double speed, boolean pauseWhenMobIdle) {
           super(wroggler, speed, pauseWhenMobIdle); //Error in first sentence happens here again
        }
    }
```

**Comment by warjort** [comment]
Your code
````
        public WrogglerAttackGoal(WrogglerEntity wroggler, double speed, boolean pauseWhenMobIdle) {
           super(wroggler, speed, pauseWhenMobIdle); //Error in first sentence happens here again
        }
````
Super constructor
````
public ZombieAttackGoal(ZombieEntity zombie, double speed, boolean pauseWhenMobIdle) {
````
WrogglerEntity is not a ZombieEntry i.e. ````wroggler instanceof ZombieEntity == false````

Try making a copy of the ZombieAttackGoal class with ZombieEntity replaced with WrogglerEntity.

**Reply by deanwean** [reply]
That also didn't work, but it is fine since I found an alternative when you replied.
Thanks for replying though!

**Comment by deanwean** [comment, marked as answer]
Instead of ZombieAttackGoal, I decided to use MeleeAttackGoal and it worked.
